packages/app/providers/swr-provider.tsx

```diff
@@ -72,6 +72,19 @@ export const SWRProvider = ({
           const maxRetryCount = config.errorRetryCount;
           const currentRetryCount = opts.retryCount;
 
+          if (error.response?.status === 401) {
+            // we only want to refresh tokens once and then bail out if it fails on 401
+            // this is to prevent infinite loops. Actually, we should logout the user but AuthProvider is not available here
+            if (currentRetryCount > 1) {
```

Comment:
is `currentRetryCount` 401 specific retry count?

Reply:
No its not, it is a general count, but the retry is happening for the same error, so it should be safe to add that specific here. If refreshToken would NOT fail, we'd return and not end up again since the timeout is coming later

Reply:
okay, but the retry count would have still gone up even if it's 404 for example right? since it's coming from `onErrorRetry` some internal counter in SWR

Reply:
Yes, the counter will increment for a 404 error, but only for that particular request. I don't believe there is a scenario where we have a 401 error that could result in a 404 on the next request, or vice versa. (While possible in theory)

However, it's possible that we might encounter a 401 error that could turn into a 500 error in the next request, which could potentially cause the retry-code to break. To check for this explicitly, we need to define a "let" outside the code and increment and reset it on success. Currently, without a bailout mechanism, we would retry 401 errors up to 8 times per hook, which caused the issue in my findings (and could be hundreds of times on the main feed).

Reply:
> but only for that particular request

Got it. I thought it was a global error request counter. My bad. LGTM ❇️